src/Storages/StorageDistributed.cpp

```diff
@@ -1265,10 +1265,19 @@ void StorageDistributed::initializeFromDisk()
     /// Make initialization for large number of disks parallel.
     ThreadPool pool(CurrentMetrics::StorageDistributedThreads, CurrentMetrics::StorageDistributedThreadsActive, CurrentMetrics::StorageDistributedThreadsScheduled, disks.size());
 
+    ThreadGroupPtr thread_group = CurrentThread::getGroup();
     for (const DiskPtr & disk : disks)
     {
         pool.scheduleOrThrowOnError([&]()
         {
+            SCOPE_EXIT_SAFE(
```

Comment:
you can probably use `scheduleFromThreadPool` from `threadPoolCallbackRunner.h`

Reply:
Done